packages/loopring_v3/circuit/Gadgets/MathGadgets.h

```diff
@@ -314,6 +461,46 @@ class AddGadget : public GadgetT
     }
 };
 
+// A - B = sub with A, B and sub >= 0
+class SubGadget : public GadgetT
```

Comment:
I'm not completely sure why Harry implemented it like that to be honest, but they are actually very similar internally (variable is broken into bits for a range check but also for a comparison) and should work the same.

Reply:
Comparison gadgets also break up the input in bits, which is the only cost for the simple range check. But the comparison gadget is a bit more expensive because it is able to compare 2 variables with each other so needs to do extra work on top of that. So the cost will be higher, but still quite similar.